common/src/main/java/com/wynntils/screens/widgets/BackButton.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.utils.McUtils;
+import net.minecraft.client.gui.components.AbstractButton;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.TextComponent;
+
+public class BackButton extends AbstractButton {
+    private final Screen backTo;
+
+    public BackButton(int x, int y, int width, int height, Screen backTo) {
+        super(x, y, width, height, new TextComponent("Back Button"));
+        this.backTo = backTo;
+    }
+
+    @Override
+    public void renderButton(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        Texture backArrow = Texture.BACK_ARROW;
+        if (this.isHovered) {
+            RenderUtils.drawTexturedRect(
```

Comment:
And it should basically only take the `Texture`, x, y, and a boolean leftHalf. 

(In fact, it might *also* be helpful to have a `drawTexturedRect` overload that takes in a Texture and extract the size and resource() automatically; this is another common pattern in this PR)

Reply:
Yes.